core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/DataTransferSupport.scala

```diff
@@ -26,23 +26,23 @@ trait DataTransferSupport extends BreakpointSupport {
     }
   }
 
-  def resumeDataTransfer()(implicit sender: ActorRef): Unit = {
+  def resumeDataTransfer(): Unit = {
     var i = 0
     while (i < output.length) {
-      output(i).resume()
+      output(i).resume()(sender)
```

Comment:
I suggest we also start easing up the writing logic. I don't like the implicit argument passing. Unless you have strong reasons to use this, please get rid of it.  If you plan to get rid of it in a separate PR, then we can do that

Reply:
I am also thinking that this entire class will be refactored when we make Messaging Module. So, the other comments that I have about the functionality of this class, I will use there